### PR TITLE
Fix SLES4SAP HANA tests

### PR DESCRIPTION
**Pull request by ldevulder**
HANA test fails on qemu backend because 'fork_bomb' test simply can't work on qemu.

The "forkbomb" test has also been moved into a separate script to ensure a better readability (should be added with `APPTESTS=xxxx,sles4sap/forkbomb`).

This commit also fix a `select_console` connection issue on IPMI backend as well as a sporadic issue that can happen during free space PV reclaims.

- Related ticket: N/A
- Needles: N/A
- Verification runs: [HANA wizard test on IPMI backend](https://openqa.suse.de/t3350406), [HANA wizard test on ppc64le](https://openqa.suse.de/t3352623)
- Regression runs: [sles4sap_scc_gnome_patterns_sapconf](https://openqa.suse.de/t3350394), [sles4sap_scc_gnome_patterns_saptune](https://openqa.suse.de/t3350527), [sles4sap_scc_gnome_saptune_v2_sapconf](https://openqa.suse.de/t3350404), [migration_offline+scc_sles4sap12sp4](https://openqa.suse.de/t3350531), [migration_offline+dvd_sles4sap12sp2_ltss](https://openqa.suse.de/t3350532)

### CODE REVIEW

File: lib/sles4sap.pm [hunk 1]
```diff
@@ -243,13 +243,12 @@ sub test_pids_max {
 }
 
 sub test_forkbomb {
-    # NOTE: Do not call this function on the qemu backend.
-    #   The first forkbomb can create 3 times as many processes as the second due to unknown bug
-    assert_script_run "curl -f -v " . autoinst_url . "/data/sles4sap/forkbomb.pl > /tmp/forkbomb.pl; chmod +x /tmp/forkbomb.pl";
+    my $script = 'forkbomb.pl';
+    assert_script_run "curl -f -v " . autoinst_url . "/data/sles4sap/forkbomb.pl -o /tmp/$script; chmod +x /tmp/$script";
```
Comment (alvarocarvajald) [inline lib/sles4sap.pm]: Why not `/data/sles4sap/$script`?

File: tests/sles4sap/wizard_hana_install.pm [hunk 2]
```diff
@@ -42,26 +43,42 @@ sub try_reclaiming_space {
...
+    # Set unit as GB, because adding TB is too much!
+    if ($unit eq 'T') {
+        $newsize *= 1024;
+        $unit = 'G';
+    }
+
+    # We can have allocated space at the end of the PV, so we need to move it before
+    my $script = 'reclaim_free_space_in_pv.sh';
+    assert_script_run "curl -f -v " . autoinst_url . "/data/sles4sap/$script -o /tmp/$script; chmod +x /tmp/$script";
+    assert_script_run "$script $device";
```
Comment (alvarocarvajald) [inline tests/sles4sap/wizard_hana_install.pm]: Shouldn't we run the script with `set -e`? I would either add a `bash -e` in this line, or better yet a `set -e` in the script itself.

File: lib/sles4sap.pm (hunk 1 above)
Comment (ldevulder) [reply lib/sles4sap.pm]: Because I forget to change this line :D

File: tests/sles4sap/wizard_hana_install.pm (hunk 2 above)
Comment (ldevulder) [reply tests/sles4sap/wizard_hana_install.pm]: Done!

In fact I'm usually don't use `set -e`, I prefer to correctly trap the error in my bash script. But in that case (bash script inside a perl script inside openQA) this make sense.

File: data/sles4sap/reclaim_free_space_in_pv.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+while (( NR_FREE > 1)); do
+  PV_LIST=$(pvs --noheadings -v --segments ${DEVICE})
+
+  # We need to remove the LV that begin at 0
+  PV_LIST=$(grep -v "${DEVICE}:0-[0-9]*" <<<"${PV_LIST}")
+
+  while IFS= read PV; do
+    if grep -q free <<<"${PV}"; then
+      # Keep the Start and SSize variables
+      START=$(awk '{print $7}' <<<"${PV}")
```
Comment (ricardobranco777) [inline data/sles4sap/reclaim_free_space_in_pv.sh]:
```suggestion
      read START SSIZE <<< $(awk '{print $7, $8}' <<<"${PV}")
```
;)

Comment (ldevulder) [reply data/sles4sap/reclaim_free_space_in_pv.sh]: Good point! Same for `PV_DEV` and `PV_SIZE` :-)

File: lib/sles4sap.pm [hunk 4]
```diff
@@ -273,48 +294,85 @@ sub test_instance_properties {
...
-    $output = script_output "pgrep -a sapstartsrv | grep -w $sid";
-    my @olines = split(/\n/, $output);
-    die "sapcontrol: wrong number of processes running after a StartService\n\n" . @olines unless (@olines == 1);
-    die "sapcontrol failed to start the service" unless ($output =~ /sapstartsrv/);
+    # Check if service is correctly started
+    $self->check_service_state('start');
+
+    # Process can take some time to initialize all
+    sleep 30;
+
```
[the remark this answers is not in the document]
Comment (ldevulder) [reply lib/sles4sap.pm]: Yes could be, I will try.

[the remark this answers is not in the document]
Comment (ldevulder) [reply lib/sles4sap.pm]: As we discussed I added the **GRAY** state but a `sleep` is still needed to be sure that `GetSystemInstanceList` can handle the requests. But I was able to lower it to 10s.

So can we merge?

**Pull request merged**